osu.Game/Overlays/Pause/PauseButton.cs

```diff
@@ -0,0 +1,241 @@
+                        },
+                        new Box
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Origin = Anchor.TopRight,
+                            Anchor = Anchor.TopRight,
+                            Width = 0.125f,
+                            ColourInfo = ColourInfo.GradientHorizontal(buttonColour, new Color4(buttonColour.R, buttonColour.G, buttonColour.B, 0f))
+                        }
+                    }
+                },
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Origin = Anchor.Centre,
+                    Anchor = Anchor.Centre,
+                    Masking = true,
+                    Children = new Drawable[]
+                    {
+                        colourContainer = new Container
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Origin = Anchor.Centre,
+                            Anchor = Anchor.Centre,
+                            Width = colourWidth,
+                            Masking = true,
+                            EdgeEffect = new EdgeEffect
+                            {
+                                Type = EdgeEffectType.Shadow,
+                                Colour = Color4.Black.Opacity(0.2f),
+                                Radius = 5
+                            },
+                            Colour = buttonColour,
+                            Shear = colourShear,
+                            Children = new Drawable[]
+                            {
+                                new Box
+                                {
+                                    EdgeSmoothness = new Vector2(2, 0),
+                                    RelativeSizeAxes = Axes.Both
+                                },
+                                new Triangles
+                                {
+                                    BlendingMode = BlendingMode.Additive,
+                                    RelativeSizeAxes = Axes.Both,
+                                    TriangleScale = 4,
+                                    Alpha = 0.05f,
+                                    Shear = -colourShear
+                                }
+                            }
+                        }
+                    }
+                },
+                spriteText = new SpriteText
+                {
+                    Text = Text,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    TextSize = 28,
+                    Font = "Exo2.0-Bold",
+                    Shadow = true,
+                    ShadowColour = new Color4(0, 0, 0, 0.1f),
+                    Colour = Color4.White,
+                }
+            });
+        }
+
+        public PauseButton()
+        {
+            
+        }
+    }
+
+    public enum PauseButtonType

```

Comment:
as mentioned in discord, best do away with this enum and subclass the buttons as necessary instead.